### DISCUSSION TITLE
How to get total edge count with `prismaNode` and `prismaConnection`?

### DISCUSSION THREAD

**Discussion by ricardo-valero** [hayes/pothos#343] [Q&A] [answered]
I have something like this

```ts
builder.prismaNode('Product', {
  include: { _count: true },
  findUnique: id => ({ id }),
  id: { resolve: product => product.id },
  fields: t => ({
    createdAt: t.expose('createdAt', { type: 'DateTime' }),
    massKg: t.expose('massKg', { type: 'Decimal', nullable: true }),
    tolerance: t.expose('tolerance', { type: 'Decimal', nullable: true }),
    name: t.exposeString('name'),
    description: t.exposeString('description', { nullable: true }),
    location: t.exposeString('location', { nullable: true }),
  }),
})

builder.queryField('products', t =>
  t.prismaConnection({
    type: 'Product',
    cursor: 'id',
    resolve: (query, parent, args, { prisma }) =>
      prisma.product.findMany({ ...query }),
  }),
)
```

I'm currently using this without issue, but I need somehow to get the total count in order to calculate the total pages in my pagination bar.

**Comment by hayes** [comment, marked as answer]
If you don't need to arguments for the field, this is pretty easy:

```ts
builder.queryFields((t) => ({
  posts: t.prismaConnection(
    {
      type: 'Post',
      cursor: 'id',
      resolve: (query) =>
        db.post.findMany({
          orderBy: { createdAt: 'desc' },
          ...query,
        }),
    },
    {
      fields: (t) => ({
        totalCount: t.int({
          resolve: () => db.post.count(),
        }),
      }),
    },
  ),
}));
```

Unfortunately the arguments for the prismaConnection field are not directly available in fields defined on the connection object.  If you need to access arguments to get the count, this won't work.  I can't think of a simple solution currently, and that might require some changes to the plugin to make that work.   If you have any suggestions on how you would want this to work let me know!

**Reply by P4sca1** [reply]
I am also struggling with implementing a `totalCount` field on a prisma connection in pothos, because the connection args are not available when adding extra fields to it. Maybe a new option could be added to let the plugin add the `totalCount` field automatically?

**Reply by hayes** [reply]
I'll get this added today, can't do it automatically, but I can add an option that should make it easy

**Reply by hayes** [reply]
Just pushed this change, it's building an publishing now, should be released in ~10 minutes

**Reply by hayes** [reply]
Can be used like this: https://github.com/hayes/pothos/blob/main/packages/plugin-prisma/tests/example/schema/index.ts#L408

**Reply by danieljvdm** [reply]
@hayes Could you add something similar to the relay connection style?  Not sure how it'd need to be implemented, but I'd like to be able to return a `total` field on my connection object. I can do so like so:
```ts
    {
      name: 'CollectionsConnection',
      fields: (t) => ({
        total: t.int({
          resolve: () => {
            return 20;
          },
        }),
      }),
    },
```

But the parent param in the resolver isn't useful nor is the `args` param since it doesn't have the arguments from the connection in there.

**Reply by hayes** [reply]
there are 2 different ways you can do it here:

https://stackblitz.com/edit/typescript-p1ftfm?file=index.ts

This shows both how to define a global totalCount field that applies to all connections, and how to define a totalCount inline for a single connection

**Reply by danieljvdm** [reply]
Wow thanks for the fast response. The `globalConnectionField` API and adding the types to the builder were what I was missing. It's working beautifully now. Thanks so much!
